### PR TITLE
Pebble FAQ

### PR DESCRIPTION
**Pull request by Ben8t**

### CODE REVIEW

**Comment by Ben8t** [comment]
The first draft can be reviewed/merged. I will add other examples/questions whenever users raise something general enough not covered yet

File: content/docs/16.faq/pebble.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+---
+title: Pebble FAQ
+---
+
+## What's Pebble ?
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:5]:
```suggestion
## What is Pebble and why do we use it at Kestra?
```

Comment (anna-geller) [reply content/docs/16.faq/pebble.md:5]: please fix the same in all titles etc like "How to reading outputs & inputs?"

you can either remove `?` or remove space

File: content/docs/16.faq/pebble.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+Pebble is a Java templating engine inspired by [Twig](https://twig.symfony.com/) and similar to the [Python Jinja Template Engine](https://palletsprojects.com/p/jinja/) syntax.
+
+In Kestra, it's used for variable operations and passing metadata between tasks.
+
+## How to reading outputs & inputs ?
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:10]:
```suggestion
## How to read outputs & inputs using Pebble
```

File: content/docs/16.faq/pebble.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+tasks:
+  - id: say_hello
+    type: io.kestra.core.tasks.log.Log
+    message: "Hello 👋, my name is {{ inputs.name }}"
+```
+
+### Read task ouputs
+
+Most of Kestra tasks expose outputs values. You can access those outputs in other tasks by using `outputs.<task_name>.<output_name>`. Every task outputs can be found the corresponding task documentation.
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:32]:
```suggestion
Most of Kestra's tasks expose output values. You can access those outputs in other tasks by using `outputs.<task_name>.<output_name>`. Every task output can be found in the corresponding task documentation.
```

File: content/docs/16.faq/pebble.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+In the example below we use the `value` outputs of the `io.kestra.core.tasks.debugs.Return` task in the downstream task.
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:34]:
```suggestion
In the example below, we use the `value` outputs of the `io.kestra.core.tasks.debugs.Return` task in the downstream task.
```

File: content/docs/16.faq/pebble.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    type: io.kestra.core.tasks.debugs.Return
+    format: "Hello 👋, my name is {{ inputs.name }}"
+
+
+  - id: can_you_repeat
+    type: io.kestra.core.tasks.log.Log
+    message: '{{ outputs.say_hello.value }}'
+```
+
+## How to format date ?
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:55]:
```suggestion
## How to format a date in Pebble
```

File: content/docs/16.faq/pebble.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+Pebble can be very useful to make small transformation on the fly - without the need to use Python or some dedicated programming language. 
+For instance, we can use the `date` function to format date values: `'{{ inputs.my_date | date("yyyyMMdd") }}'`
+
+You can find more documentation on the `date` function [in this documentation](../05.developer-guide/03.variables/03.filter/date.md)
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:60]:
```suggestion
You can find more documentation on the `date` function [in the Variables section of the Developer Guide](../05.developer-guide/03.variables/03.filter/date.md)
```
avoid "here"/"this" in docs, when links break, you won't know what this link is

File: content/docs/16.faq/pebble.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+
+### Coalesce operator for trigger or manual execution
+
+Most of the time, a flow will be triggered automatically. Either on schedule or based on external events. It’s common to use the date of the execution to process the corresponding data and make the flow dependent on time. 
+
+With Pebble you can use the `trigger.date` to get the date of the executed trigger.
+Still, sometimes you want to manually execute a flow. Then the `trigger.date` variable won’t work anymore. For this you can use the `execution.startDate` variable that returns the execution start date.
+
+It’s easy to make both behavior work without overhead using the coalesce operator `??`
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:69]:
```suggestion
To support both use cases, use the coalesce operator `??`. The example below shows how to apply it in a flow.
```

File: content/docs/16.faq/pebble.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    cron: "* * * * *"
+```
+
+## How to parse objects & list ?
+
+Sometimes outputs are made of nested objects or list. To parse those elements, you will need to use `jq` most of the time.
+
+jq is like sed for JSON data - you can use it to slice, filter, map and transform structured data with the same ease that sed, awk, grep and similar commands let you play with text.
+
+For example, giving the following flow data
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:92]:
```suggestion
Sometimes, outputs return nested objects or lists. To parse those elements, you may leverage `jq`. You can use jQuery to slice, filter, map, and transform structured data with the same ease that `sed`, `awk`, `grep`, and similar Linux commands let you manipulate strings.

Consider the following flow:
```

File: content/docs/16.faq/pebble.md [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+
+```
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:94]:
```suggestion
```yaml
```

File: content/docs/16.faq/pebble.md [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+  - id: hello
+    type: io.kestra.core.tasks.log.Log
+    message: "{{ inputs.data }}"
+```
+
+The expression `{{ inputs.data.value }}` will return the list `[1, 2, 3]`
+
+The expression `{{ inputs.data.value | jq(".[1]") | first }}` will return `2`.
+
+`jq(".[1]")` access the second value of the list and return an array with one element, we then use `first` to access the value itself.
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:114]:
```suggestion
`jq(".[1]")` accesses the second value of the list and returns an array with one element. We then use `first` to access the value itself.
```

File: content/docs/16.faq/pebble.md [hunk 11]
```diff
@@ -0,0 +1,149 @@
...
+
+> Note: we could have used `{{ inputs | jq(".data.value[1]") | first }}`, jq allows to parse any object in Kestra context.
+
+You can play with the [Eval Expression](../04.user-interface-guide/04-executions.md) button in the outputs tab of a Flow execution. It's really helpful to troubleshoot more complex object parsing.
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:118]:
```suggestion
You can troubleshoot complex Pebble expressions using the [Eval Expression](../04.user-interface-guide/04-executions.md) button in the outputs tab of a Flow execution. It's helpful to validate how complex objects will be parsed.
```

File: content/docs/16.faq/pebble.md [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+
+
+## How to use conditions ?
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:121]:
```suggestion
## How to use conditions in Pebble
```

File: content/docs/16.faq/pebble.md [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+
+In some tasks like the If or Switch tasks, you will need to provide some conditions. You can use the Pebble syntax to use previous task outputs in those conditions:
```
Comment (anna-geller) [inline content/docs/16.faq/pebble.md:123]:
```suggestion
In some tasks, such as the `If` or `Switch` tasks, you will need to provide some conditions. You can use the Pebble syntax to use previous task outputs within those conditions:
```

**Pull request merged**